### ISSUE TITLE
[BUG 4.0.7] Using tracked with co-location results in double setting of template

### ISSUE THREAD

**Issue by runspired** [ember-cli/ember-cli-htmlbars#339]
This issue goes away by pinning to 4.0.6

Commit by commit reproduction from the 3.13 release blog post instructions can be found here: https://github.com/runspired/ember-tracked-component-bug

Install and run tests to see:

```
vendor.js:37145 Uncaught (in promise) Error: Assertion Failed: Cannot call `setComponentTemplate` multiple times on the same class (`class TickerComponent extends _component.default {
```

tracing to source of error you will see:

![](https://cdn.discordapp.com/attachments/608346628163633192/635098073282117662/unknown.png)

**Comment by runspired** [comment]
From @rwjblue 

```
I know what is wrong
If you file the issue, I’ll try to explain there. But tldr, we fixed a bug with TS by adding an additional hook to our plugin
And now we are hitting it twice (both for ExportDefaultStatement and ExportStatement with a default) causing us to emit the setComponentTemplate stuff twice
 
rwjblueToday at 4:50 PM
^ is almost certainly because of a Babel plugin ordering issue (where our decorator transform causes us to hit export { SomeName as default } afger we have seen the export default SomeName)
:weary: sorry folks
```

**Comment by rwjblue** [comment]
Thank you for reporting! I think this is caused by the changes that landed in 4.0.7 (adding support for transforming `export { Foo as default }` statements) when used along with Babel’s decorator transforms. 

Basically, I think we hit our ExportDefault hook before Babel has transformed the class, then it transforms it to use the `export { Foo as default }` syntax, which we _also_ hook and then write the `setComponentTemplate` invocation a second time. 

The fix, is to track that we have already added the `setComponentTemplate` and avoid adding it again.

**Comment by rwjblue** [comment]
Submitted fix in #340

**Comment by rwjblue** [comment]
The fix is released in [v4.0.8](https://github.com/ember-cli/ember-cli-htmlbars/releases/tag/v4.0.8).
